CacheSim/CacheSim.h

```diff
@@ -28,25 +28,63 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <stdint.h>
 #include <stdio.h>
+#if defined(_MSC_VER)
 #include <Windows.h>
+#ifndef IG_CACHESIM_API
+#define IG_CACHESIM_API __declspec(dllimport)
+#endif
+#else
+#include <dlfcn.h>
+#include <unistd.h>
+#define IG_CACHESIM_API
```

Comment:
Suggest this becomes `__attribute__((visibility(default)))` or whatever the incantation is so the solib can be compiled with hidden visibility (speeds up `dlopen`)